### DISCUSSION TITLE
Need help - some of my students can see repos belonging to other students

### DISCUSSION THREAD

**Discussion by shenoy1** [community/community#119974] [GitHub Education] [answered]
### Hi everyone,

Hello -- We are encountering very odd behavior from Github Classroom.

Problem 1:  I have a group assignment for the class that is done in groups of 2.  When students accept the github classroom invite link for the assignment, they create a team for their new forked repo.  Their team is getting added both to the forked repo and the PARENT repo (created by us to give out the assignment). As a result, any subsequent student who accepts the invite gets a forked repo with their team and all students teams who mistakenly got added to the parent.  This means that students can see all repos that were forked after they created their repo.     The repos are private but other students (their teams) are getting added to a private repo.  We are having to manually clean up this issue.

Problem 2:  We are seeing random github accounts getting added as outside collaborators to the parent repo holding the assignment.  Our repo is private and we have neither invited nor accepted anyone to our repos. Yet, we have up to 60 outside collaborators show up on our assignment repos with read access.  How is this even happening? 

Here is an example parent repo with both problems: https://github.com/umass-cs677-current/spring24-lab2/settings/access 

This repo has 60 strangers as outside collaborators and 3 student teams who got added to the parent (and then got access to many other student repos that were forked subsequently)

Any help would be highly appreciated.  Also, this issue is only seen for group assignments. All individual assignments are working normally.

**Comment by hochleitner** [comment]
I'm cross-referencing discussions from the Global Campus Teacher discussions here. 

Problem 1 might be related to this one: https://github.com/community/Global-Campus-Teachers/discussions/329

Problem 2 is this one, which should actually be already resolved (but maybe it's not): https://github.com/community/Global-Campus-Teachers/discussions/317

**Comment by kevinwortman** [comment]
I am experiencing problem 1 as well. Here is another example of an affected parent repo: https://github.com/cpsc-spring-2024/cpsc-120-prompt-lab-08

**Comment by RyanHecht** [comment, marked as answer]
Hi @shenoy1, thanks for the report!

As @hochleitner mentioned, Problem 2 should have been resolved by us and we _should_ have removed the invalid outside collaborators -- our team will look into why this wasn't the case for you.

Problem 1 seems like a novel bug; I've flagged it to the team and we'll have a look.

**Reply by kevinwortman** [reply]
Thanks!

**Comment by shenoy1** [comment]
The bug in problem 1 can be reproduced as follows:

- create a group assignment with a starter code repo.
- Student 1 accepts invitation from google classroom and creates a new team when accepting the assignment.

- Classroom forks a repo for student 1. The new team gets added  to the parent/starter code repo.

- Student 2 accepts invitation and creates a new team.  Classroom forks a new repo, which inherits teams from the starter code repo.
Now, student 1’s team can see student 2’s repository.

**Comment by famosab** [comment]
Hey I want to remove the read access for the repos of other teams in the same classroom. How can I do that?

**Reply by hochleitner** [reply]
You have to go to your classroom's organization, there you will see all your teams. Open a team and you see their repos. Now you can remove the unwanted access.

Depending on the number of assignments and teams this is very tedious work. I had hoped that @RyanHecht and the team could somehow restore it to the correct state.

**Reply by famosab** [reply]
I did that now. Thanks! Is there also a way to disable that the students can open PRs to the original Assignment repo?

**Comment by Rifahad** [comment]
Problem is this one, which should actually be already resolved (but maybe it's not): https://github.com/community/Global-Campus-Teachers/discussions/317

**Reply by Rameeskc** [reply]
got it
